src/sgame/botlib/bot_nav_edit.cpp

```diff
@@ -276,6 +288,96 @@ void Cmd_AddConnection( gentity_t *ent )
 
 	arg = Cmd_Argv( 1 );
 
+	if ( cmd.enabled )
+	{
+		if ( !Q_stricmp( arg, "start" ) )
+		{
+			if ( argc < 3 )
+			{
+				Log::Notice( usage );
+				return;
+			}
+
+			arg = Cmd_Argv( 2 );
+
+			if ( !Q_stricmp( arg, "oneway" ) )
+			{
+				cmd.pc.dir = 0;
+			}
+			else if ( !Q_stricmp( arg, "twoway" ) )
+			{
+				cmd.pc.dir = 1;
+			}
+			else
+			{
+				Log::Notice( "Invalid argument for direction, specify oneway or twoway" );
+				return;
+			}
+
+			if ( GetPointPointedTo( cmd.nav, cmd.pc.start ) )
+			{
+				cmd.pc.area = DT_TILECACHE_WALKABLE_AREA;
+				cmd.pc.flag = POLYFLAGS_WALK;
+				cmd.pc.userid = 0;
+				cmd.pc.radius = navconRadius.Get();
+				cmd.offBegin = true;
+			}
+		}
+		else if ( !Q_stricmp( arg, "end" ) )
+		{
+			if ( !cmd.offBegin )
+			{
+				return;
+			}
+
+			if ( GetPointPointedTo( cmd.nav, cmd.pc.end ) )
+			{
+				cmd.nav->process.con.addConnection( cmd.pc );
+
+				rBounds box;
+				box.addPoint( cmd.pc.start );
+				box.addPoint( cmd.pc.end );
+
+				box.mins[ 1 ] -= 10;
+				box.maxs[ 1 ] += 10;
+
+				// rebuild affected tiles
+				dtCompressedTileRef refs[ 32 ];
+				int tc = 0;
+				cmd.nav->cache->queryTiles( box.mins, box.maxs, refs, &tc, 32 );
+
+				for ( int k = 0; k < tc; k++ )
+				{
+					cmd.nav->cache->buildNavMeshTile( refs[ k ], cmd.nav->mesh );
+				}
+
+				cmd.offBegin = false;
+			}
+		}
+	}
+	else
+	{
+		Log::Notice( usage );
+	}
+}
+
+void Cmd_PutCon_f( gentity_t *ent )
+{
+	if ( !CheckHost( ent ) ) return;
+	const char usage[] = "Usage: putcon start <oneway|twoway> \n"
+	                     " putcon end\n"
```

Comment:
It's really odd the use `start` then `end`. Maybe start/stop? or maybe rather begin/end?

Reply:
That's using the same syntax as already existing code. It's unlikely people put bindings in their configs, so this can be changed, sure.
I don't really think this is that important, what I would like in the end is to never again have to write those damned commands, by having proper bindings.
What is important, though, is to provide the feature.